### ISSUE TITLE
Strange output for a null size file alone in a directory.

### ISSUE THREAD

**Issue by Tracnac** [gilbertchen/duplicacy-cli#50]
Hello,

Don't know if it's a bug, but the output message is not very nice for a null size file... (Debug ?!)

Every things looks good when a null size file plus another not null... (Even if I don't really like the output of  the null size file...)

````bash
[root@8ad75214293c home]# uname -a
Linux 8ad75214293c 4.9.13-moby #1 SMP Sat Mar 25 02:48:44 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
[root@8ad75214293c home]# cat /etc/redhat-release
CentOS Linux release 7.3.1611 (Core)
[root@8ad75214293c home]#

[root@8ad75214293c home]# rm -r .duplicacy ; rm -r /mnt/data/backup ; mkdir /mnt/data/backup
[root@8ad75214293c home]# duplicacy_linux init 0 /mnt/data/backup
/home will be backed up to /mnt/data/backup with id 0
[root@8ad75214293c home]# ls -l
total 0
[root@8ad75214293c home]# touch a
[root@8ad75214293c home]# echo 'Hello the world' > b
[root@8ad75214293c home]# duplicacy_linux backup
Storage set to /mnt/data/backup
No previous backup found
Indexing /home
Packed b (16)
Backup for /home at revision 1 completed
[root@8ad75214293c home]# duplicacy_linux list -files
Storage set to /mnt/data/backup
Snapshot 0 revision 1 created at 2017-04-21 14:54 -hash
 0 2017-04-21 14:53:58                                                                  a
16 2017-04-21 14:54:06 00d0eaf33eedf0a80f68d75b6c4b8fe20be8ed2a978798458db30271608c4326 b
Files: 2, total size: 16, file chunks: 1, metadata chunks: 3
[root@8ad75214293c home]# duplicacy_linux cat -r 1 a
File a has mismatched hashes:  vs 0e5751c026e543b2e8ab2eb06099daa1d1e5df47778f7787faab45cdf12fe3a8
File a is corrupted in snapshot 0 at revision 1
[root@8ad75214293c home]# duplicacy_linux cat -r 1 b
Hello the world

[root@8ad75214293c home]#
````

Now the case :

```bash
[root@8ad75214293c home]# rm -r .duplicacy ; rm -r /mnt/data/backup ; mkdir /mnt/data/backup
[root@8ad75214293c home]# duplicacy_linux init 0 /mnt/data/backup
/home will be backed up to /mnt/data/backup with id 0
[root@8ad75214293c home]# ls -l
total 0
[root@8ad75214293c home]# touch a
[root@8ad75214293c home]# duplicacy_linux backup
Storage set to /mnt/data/backup
No previous backup found
Indexing /home
Backup for /home at revision 1 completed
[root@8ad75214293c home]# duplicacy_linux list -files
Storage set to /mnt/data/backup
Snapshot 0 revision 1 created at 2017-04-21 15:01 -hash
0 2017-04-21 15:01:31                                                                  a
Files: 1, total size: 0, file chunks: 1, metadata chunks: 3
[root@8ad75214293c home]# duplicacy_linux cat -r 1 a
runtime error: index out of range
goroutine 1 [running]:
runtime/debug.Stack(0x22, 0x0, 0x0)
        /usr/local/go/src/runtime/debug/stack.go:24 +0x79
runtime/debug.PrintStack()
        /usr/local/go/src/runtime/debug/stack.go:16 +0x22
github.com/gilbertchen/duplicacy.CatchLogException()
        /Users/chgang/zincbox/go/src/github.com/gilbertchen/duplicacy/duplicacy_log.go:166 +0xf4
panic(0x8c7de0, 0xc420010080)
        /usr/local/go/src/runtime/panic.go:458 +0x243
github.com/gilbertchen/duplicacy.(*SnapshotManager).RetrieveFile(0xc420512000, 0xc4200ba2d0, 0xc42007a100, 0xc4200efbd8, 0xc42007a100)
        /Users/chgang/zincbox/go/src/github.com/gilbertchen/duplicacy/duplicacy_snapshotmanager.go:1041 +0x6c5
github.com/gilbertchen/duplicacy.(*SnapshotManager).PrintFile(0xc420512000, 0xc420011f1f, 0x1, 0x1, 0x7ffc1037c918, 0x1, 0x0)
        /Users/chgang/zincbox/go/src/github.com/gilbertchen/duplicacy/duplicacy_snapshotmanager.go:1120 +0x360
main.printFile(0xc4200e65a0)
        /Users/chgang/zincbox/go/src/github.com/gilbertchen/duplicacy/main/duplicacy_main.go:819 +0x3b0
github.com/gilbertchen/cli.Command.Run(0x949ede, 0x3, 0x0, 0x0, 0x0, 0x0, 0x0, 0x969419, 0x53, 0x0, ...)
        /Users/chgang/zincbox/go/src/github.com/gilbertchen/cli/command.go:160 +0xacd
github.com/gilbertchen/cli.(*App).Run(0xc4200e6360, 0xc42000a0f0, 0x5, 0x5, 0x0, 0x0)
        /Users/chgang/zincbox/go/src/github.com/gilbertchen/cli/app.go:179 +0x919
main.main()
        /Users/chgang/zincbox/go/src/github.com/gilbertchen/duplicacy/main/duplicacy_main.go:1639 +0x4722
[root@8ad75214293c home]#
```

I'm keeping watching for any additionnal help...
Thanks Yvan

Sorry if my english is not very clear.

**Comment by gilbertchen** [comment]
This is a bug.  It doesn't handle 0-size files correctly.

**Comment by gilbertchen** [comment]
This should be fixed by version 2.0.0.

**Comment by Tracnac** [comment]
Thanks for your feedback.
